config/crds/servicemesh.kubesphere.io_strategies.yaml

```diff
@@ -259,8 +259,10 @@ spec:
                                     additionalProperties:
                                       description: Describes how to match a given string in HTTP headers. Match is case-sensitive.
                                       type: object
+                                      x-kubernetes-preserve-unknown-fields: true
```

Comment:
We should avoid change CRD YAML directly because it will be overwritten by controller-gen.

Reply:
As usually, we should avoid change directly, but istio use [cue-gen](https://github.com/istio/tools/blob/master/cmd/cue-gen/README.md#generate-crds) tool to generate CRD. So there is something different about the markers from controller-gen. So we should modify it by hand, because controller-gen cannot generate it correctlly.

Reply:
https://github.com/kubesphere/kubesphere/pull/3795 Does this PR help?

Reply:
no use. The markers only take effect in the root object. But this place should do in the lower object.https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#controlling-pruning

Reply:
I think we need to use cue-gen to regenerate crd files again instead of patching it manually, cause we don't know if we have modified the files completely. Need more suggestions from @zackzhangkai.

Reply:
cue-gen is a tool just from istio. we just have two istio crds, no need to use this tool.  Indeed, we no need to focus on the tools, they are just help us to generate manifest, which based on [open API v3]([https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.0.md#schemaObject](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.0.md#schemaObject)). In the fact, we just to take a full test on our products, that's OK, I think.

Reply:
I will create another issue to track the test on service mesh.